### PR TITLE
Removes sad face when source+goto

### PR DESCRIPTION
**Pull request by lukaszsobek**
Associated Issue: #5428

- Removes sad face when using source+goto queries in QOM search
- handles source goto queries

Vid:
removes-sad-goto

### CODE REVIEW

File: src/components/QuickOpenModal.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ export class QuickOpenModal extends Component<Props, State> {
       const results = this.props.sources;
       return this.setState({ results });
     }
-    if (this.isGotoSourceQuery()) {
+    if (this.isGotoSourceQuery() || this.isSourcesQuery()) {
```
Comment (lukaszsobek) [inline src/components/QuickOpenModal.js]: Not really sure we need such a rigid separation between `isGotoSourceQuery` and `isSourcesQuery` as `isGotoSourceQuery` seems more like a subtype of `isSourcesQuery` than a fully fledged equivalent. Then there is `isSourceSearch`, which should have been used above in checking the type in the scenario we have now.

Comment (jasonLaster) [inline src/components/QuickOpenModal.js]: hmm not sure what this does. can you add a unit test?

Comment (lukaszsobek) [reply src/components/QuickOpenModal.js]: Sure, this causes the error emoji to disappear if the source has a `:`, but it is only half of the issue - the other half is that when the input is in `SourcesQuery` mode it does not consider the location suffix.

File: src/components/QuickOpenModal.js [hunk 2]
```diff
@@ -111,18 +111,10 @@ export class QuickOpenModal extends Component<Props, State> {
...
-      const [baseQuery] = query.split(":");
-      const results = filter(this.props.sources, baseQuery);
-      this.setState({ results });
-    } else {
-      const results = filter(this.props.sources, query);
-      this.setState({ results });
-    }
+    const { sources } = this.props;
+    const results =
+      query == "" ? sources : filter(sources, query.split(":")[0]);
```
Comment (jasonLaster) [inline src/components/QuickOpenModal.js]: lets move this to its own function `query.split(":")[0]`

`filter(sources, dropGoto(query))`

i dont like the term drop but it is nice to think about the language we use

**Comment by darkwing** [comment]
Part of me wonders if this is too simplistic.  I've seen ":" used in URLs for files before, plus we have our own ":formatted" syntax. Maybe we should try to make it smarter, looking for :[digits]?

**Pull request merged**